### PR TITLE
Gossip Metrics tracked by Agent instead of Cert

### PR DESCRIPTION
**Pull request by neonphog**
### Summary

- `initiate` and `accept` sharded gossip requests now exchange local agent infos
- these local agent infos are used to key gossip metrics based on agent pub key instead of tls cert

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs [hunk 1]
```diff
@@ -108,12 +140,31 @@ impl ShardedGossipLocal {
 
         let mut gossip = Vec::with_capacity(3);
 
+        let mut agent_list = Vec::new();
+        for agent in local_agents.iter() {
+            if let Ok(Some(info)) = self
+                .evt_sender
+                .get_agent_info_signed(GetAgentInfoSignedEvt {
```
Comment (freesig) [inline crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs:147]: Again I would do this a one call as it's all the local agents. Although I guess it depends if we are optimizing for the holo usecase or the one agent per space per conductor?

**Review by freesig — APPROVED** [review]
Looks good, just a question around the amount of calls to agent info although I guess it depends what usecase we are optimizing for so it could go either way.

File: crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs [hunk 2]
```diff
@@ -35,14 +35,45 @@ impl ShardedGossipLocal {
 
         let id = rand::thread_rng().gen();
 
+        let mut agent_list = Vec::new();
+        for agent in local_agents.iter() {
+            if let Ok(Some(info)) = self
+                .evt_sender
+                .get_agent_info_signed(GetAgentInfoSignedEvt {
```
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs]: :+1: updated

File: crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs (hunk 1 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs]: :+1: updated

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**